src/api.ts

```diff
@@ -573,7 +581,21 @@ async function injectAndRecordTx(
   status: number
 }> {
   const { raw } = tx
-  const { baseUrl } = getBaseUrl()
+  let nodeIpPort: string, baseUrl: string
+
+  // Initialize them with some values({ nodeIpPort, baseUrl } = getBaseUrl())
+
+  if (config.enableBlacklistingIP) {
+    // Continuously reassign nodeIpPort and baseUrl until a non-blacklisted one is found
+    let entry
+
+    do {
+      ;({ nodeIpPort, baseUrl } = getBaseUrl())
```

Comment:
The getBaseUrl() should be outside the config.enableBlacklistingIP. Missed it during the team review.